src/main/java/org/commcare/core/network/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,52 @@
+package org.commcare.core.network;
+
+import java.io.IOException;
+
+import javax.annotation.Nullable;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class AuthenticationInterceptor implements Interceptor {
+
+    @Nullable
+    private String credential;
+
+    public AuthenticationInterceptor(@Nullable String credential) {
```

Comment:
This constructor is now only ever called with the parameter as `null`, so can we just make it an empty constructor?